src/Gui/ElideCheckBox.cpp

```diff
@@ -0,0 +1,68 @@
+/***************************************************************************
+ *   Copyright (c) 2025 Alfredo Monclus <alfredomonclus@gmail.com>         *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 51 Franklin Street,      *
+ *   Fifth Floor, Boston, MA  02110-1301, USA                              *
+ *                                                                         *
+ ***************************************************************************/
+
+// This custom widget adds the missing ellipsize functionality in QT5
+
```

Comment:
In order to work with MSVC when PCH are enabled, all cpp files must begin with:
```suggestion

#include "PreCompiled.h"
```